src/Language/Pirouette/PlutusIR/ToTerm.hs

```diff
@@ -35,6 +36,13 @@ import Data.Function (on)
 
 -- * Translating 'PIR.Type's and 'PIR.Term's
 
+-- [HACK: Translation of 'Bool']
+-- Many PIR files define their own 'Bool' type as an ADT
+-- with their own 'True' and 'False'. However, this breaks
+-- our translation, since this 'Bool' is *not* the built-in.
+-- So instead we transform any usage of a 'PIRType' with
+-- 'Bool' into the built-in one.
```

Comment:
How do you reckon doing this as a separate `PIR` transformation? My reasoning for this is that this code is complex enough to split any special cases out to a separate pass/function/module/etc.

Sure, this can be done separately later.